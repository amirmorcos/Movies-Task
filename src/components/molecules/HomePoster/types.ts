import { StyleProp, ViewStyle } from "react-native";

export type HomePosterProps = {
  path: string;
  overrideContainerStyle?: StyleProp<ViewStyle>;
};
