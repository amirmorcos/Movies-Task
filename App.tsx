/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import { NavigationContainer } from "@react-navigation/native";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import BottomNavigation from "navigation/BottomNavigation";
import React from "react";

const queryClient = new QueryClient();

function App(): React.JSX.Element {
  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>
        <BottomNavigation />
      </NavigationContainer>
    </QueryClientProvider>
  );
}

export default App;
