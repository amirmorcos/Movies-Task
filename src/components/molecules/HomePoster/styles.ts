import { StyleSheet } from "react-native";
import { s, vs } from "react-native-size-matters";

export default StyleSheet.create({
  container: {
    width: s(140),
    height: vs(200),
    overflow: "hidden",
    borderRadius: 10,
  },
  image: {
    flex: 1,
    width: undefined,
    height: undefined,
  },
});
