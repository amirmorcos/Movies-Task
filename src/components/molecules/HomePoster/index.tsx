import React from "react";
import { Image, View } from "react-native";
import { HomePosterProps } from "./types";
import styles from "./styles";
import { useMovie } from "hooks/useMovie";
import FastImage from "react-native-fast-image";

const HomePoster = ({ path, overrideContainerStyle }: HomePosterProps) => {
  const { getMoviePoster } = useMovie();
  return (
    <View style={[styles.container, overrideContainerStyle]}>
      <FastImage
        source={{
          uri: getMoviePoster(path),
        }}
        style={styles.image}
        resizeMode="stretch"
      />
    </View>
  );
};

export default HomePoster;
