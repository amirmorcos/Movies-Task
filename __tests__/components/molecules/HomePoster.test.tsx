import { QueryClientProvider } from "@tanstack/react-query";
import { queryClient } from "../../../App";
import HomePoster from "molecules/HomePoster";
import React from "react";
import renderer from "react-test-renderer";

jest.mock("react-native-vector-icons/FontAwesome6", () => "Icon");
jest.mock("react-native-vector-icons/FontAwesome", () => "Icon");
jest.mock("react-native-size-matters", () => ({
  s: jest.fn(),
  vs: jest.fn(),
  moderateScale: jest.fn(),
}));
jest.mock("@react-navigation/bottom-tabs", () => ({
  createBottomTabNavigator: jest.fn(),
}));

jest.mock("react-native-fast-image", () => "FastImage");

test("renders correctly", () => {
  const tree = renderer
    .create(
      <QueryClientProvider client={queryClient}>
        <HomePoster path="https://image.tmdb.org/t/p/w370_and_h556_bestv2/dcnSWFCtk4b2aIzkhq6IDbzoIf1.jpg" />
      </QueryClientProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
