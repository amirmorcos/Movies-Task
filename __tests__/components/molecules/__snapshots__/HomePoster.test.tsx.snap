// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "borderRadius": 10,
        "height": undefined,
        "overflow": "hidden",
        "width": undefined,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "flex": 1,
          "height": undefined,
          "width": undefined,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="stretch"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
